packages/react-positioning/src/middleware/intersectionObserver.ts

```diff
@@ -0,0 +1,22 @@
+import { detectOverflow, Middleware } from '@floating-ui/dom';
+
+export function intersectionObserver(): Middleware {
```

Comment:
Let's name this middleware differently as "Intersection Observer" has strong connection with matching DOM API, https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API.